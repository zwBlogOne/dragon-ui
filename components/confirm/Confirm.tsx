import React, { Component } from 'react';
import Modal from '../modal';
import Button from '../button';
import Icon from '../icon';
import PropsType from './PropsType';

class Confirm extends Component<PropsType, any> {
  static defaultProps = {
    prefixCls: 'ui-confirm',
    message: '',
    width: 270,
    onOk: () => {},
    onCancel: () => {},
  };

  render() {
    const {
      prefixCls,
      message,
      okText,
      cancelText,
      onOk,
      width,
      visible,
      onCancel,
      locale,
    } = this.props;
    return (
      <Modal width={width} visible={visible}>
        <Modal.Body>
          <div className={prefixCls}>
            <Icon type="question-round" />
            <span>{message}</span>
          </div>
        </Modal.Body>
        <Modal.Footer>
          <Button onClick={onCancel}>{cancelText || locale!.cancel}</Button>
          <Button theme="success" onClick={onOk}>
            {okText || locale!.confirm}
          </Button>
        </Modal.Footer>
      </Modal>
    );
  }
}

export default Confirm;
