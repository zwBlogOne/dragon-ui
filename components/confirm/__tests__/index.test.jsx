import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Confirm from '../index';

describe('Confirm', () => {
  it('renders basic Confirm correctly', () => {
    const onOk = jest.fn();
    const onCancel = jest.fn();
    const wrapper = mount(
      <Confirm onOk={onOk} onCancel={onCancel} message="确认删除吗？" />
    );
    wrapper.setProps({ visible: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
