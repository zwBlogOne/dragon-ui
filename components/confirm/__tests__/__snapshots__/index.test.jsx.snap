// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders basic Confirm correctly 1`] = `
<LocaleReceiver
  message="确认删除吗？"
  onCancel={[Function]}
  onOk={[Function]}
  visible={true}
>
  <ConfirmExtension
    locale={
      Object {
        "cancel": "取消",
        "confirm": "确定",
      }
    }
    localeCode="zh-cn"
    message="确认删除吗？"
    onCancel={[Function]}
    onOk={[Function]}
    prefixCls="ui-confirm"
    visible={true}
    width={270}
  >
    <Modal
      animationDuration={300}
      animationType="zoom"
      isRadius={false}
      isRound={false}
      minWidth={270}
      onMaskClick={[Function]}
      prefixCls="ui-modal"
      visible={true}
      width={270}
    >
      <div
        className="ui-modal fade-enter"
        onClick={[Function]}
        style={
          Object {
            "MozAnimationDuration": "300ms",
            "OAnimationDuration": "300ms",
            "WebkitAnimationDuration": "300ms",
            "animationDuration": "300ms",
            "msAnimationDuration": "300ms",
            "position": "fixed",
          }
        }
      >
        <div
          className="ui-modal-wrapper"
        >
          <div
            className="ui-modal-dialog zoom-enter"
            onClick={[Function]}
            style={
              Object {
                "MozAnimationDuration": "300ms",
                "OAnimationDuration": "300ms",
                "WebkitAnimationDuration": "300ms",
                "animationDuration": "300ms",
                "minWidth": 270,
                "msAnimationDuration": "300ms",
                "width": 270,
              }
            }
          >
            <ModalBody
              height="auto"
              prefixCls="ui-modal"
              style={Object {}}
            >
              <div
                className="ui-modal-body"
                style={
                  Object {
                    "height": "auto",
                  }
                }
              >
                <div
                  className="ui-confirm"
                >
                  <Icon
                    className=""
                    onClick={[Function]}
                    prefixCls="ui-icon"
                    style={Object {}}
                    theme="default"
                    type="question-round"
                  >
                    <i
                      className="ui-icon ui-icon-question-round theme-default"
                      onClick={[Function]}
                      style={Object {}}
                    />
                  </Icon>
                  <span>
                    确认删除吗？
                  </span>
                </div>
              </div>
            </ModalBody>
            <ModalFooter
              prefixCls="ui-modal"
              style={Object {}}
            >
              <div
                className="ui-modal-footer"
                style={Object {}}
              >
                <Button
                  className={null}
                  isActive={false}
                  isBlock={false}
                  isCircle={false}
                  isDisabled={false}
                  isFocus={false}
                  isLoading={false}
                  isRadius={false}
                  isRound={false}
                  onClick={[Function]}
                  prefixCls="ui-button"
                  size={null}
                  style={Object {}}
                  theme="default"
                  type="button"
                >
                  <button
                    className="ui-button theme-default"
                    disabled={false}
                    onClick={[Function]}
                    style={Object {}}
                    type="button"
                  >
                    取消
                  </button>
                </Button>
                <Button
                  className={null}
                  isActive={false}
                  isBlock={false}
                  isCircle={false}
                  isDisabled={false}
                  isFocus={false}
                  isLoading={false}
                  isRadius={false}
                  isRound={false}
                  onClick={[Function]}
                  prefixCls="ui-button"
                  size={null}
                  style={Object {}}
                  theme="success"
                  type="button"
                >
                  <button
                    className="ui-button theme-success"
                    disabled={false}
                    onClick={[Function]}
                    style={Object {}}
                    type="button"
                  >
                    确定
                  </button>
                </Button>
              </div>
            </ModalFooter>
          </div>
        </div>
      </div>
    </Modal>
  </ConfirmExtension>
</LocaleReceiver>
`;
